### PR TITLE
MongoDb - add support for non-ObjectId _id fields

### PR DESCRIPTION
**Pull request by colesnodgrass**
## What
- add support for non-ObjectId `_id` fields
    - currently supports: `ObjectId`, `String`, `Int`, `Long`
- add `IdType` enum for managing `_id` type support
    - I am open to suggestions for renaming `IdType`
- add `IdType` to `MongoDbStreamState` record class
- throw an exception if multiple types of `_id` fields exist within a collection
- throw an exception if an unsupported `_id` type field exists within a collection
- refactored `createStateManager` to reduce number of nested blocks of code

## TODO
- [x] set the `IdType` correctly in the `MongoDbStateIterator`
- [x] remove `MongoDbStreamStates`

### CODE REVIEW

File: airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/mongodb/MongoDbDebeziumStateUtil.java [hunk 1]
```diff
@@ -79,7 +81,7 @@ public static JsonNode formatState(final String database, final String replicaSe
     final List<Object> key = generateOffsetKey(database, replicaSet);
 
     final Map<String, Object> value = new HashMap<>();
-    value.put(MongoDbDebeziumConstants.OffsetState.VALUE_SECONDS, timestamp.getTime());
+    value.put(VALUE_SECONDS, timestamp.getTime());
```
Comment (jdpgrailsdev) [inline airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/mongodb/MongoDbDebeziumStateUtil.java:84]: Nit:  I'm not sure what is going on with Git and IntelliJ, but this keeps yo-yo'ing between using the fully qualified name and a static import.  All of the other constants referenced from that class in this class are using the fully-qualified name, if you want to undo/put this one back so they are consistent.

File: airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/InitialSnapshotHandler.java [hunk 2]
```diff
@@ -85,4 +101,37 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIterators(
...
+    // {
+    // $group : {
+    // _id : { $type : "$_id" },
+    // count : { $sum : 1 }
+    // }
+    // }
+    // ])
+    collection.aggregate(List.of(
+        Aggregates.group(
+            new Document("_id", new Document("$type", "$_id")),
```
Comment (jdpgrailsdev) [inline airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/InitialSnapshotHandler.java:123]: Nit:  We have a constant for `_id` in `MongoConstants`.

**Review by jdpgrailsdev — APPROVED** [review]
:shipit: LGTM with a couple of very minor nits

File: airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/mongodb/MongoDbDebeziumStateUtil.java (hunk 1 above)
Comment (colesnodgrass) [reply airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/mongodb/MongoDbDebeziumStateUtil.java:84]: I've fixed this a few times already, but it seems that every time I merge the base branch into this branch it undoes it 😞

**Pull request merged**